analyzers/analysistest/analysistest.go

```diff
@@ -0,0 +1,269 @@
+package analysistest
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/deepsourcelabs/deepsource-go/analyzers/types"
+	sitter "github.com/smacker/go-tree-sitter"
+	"github.com/smacker/go-tree-sitter/bash"
+	"github.com/smacker/go-tree-sitter/c"
+	"github.com/smacker/go-tree-sitter/cpp"
+	"github.com/smacker/go-tree-sitter/csharp"
+	"github.com/smacker/go-tree-sitter/css"
+	"github.com/smacker/go-tree-sitter/elm"
+	"github.com/smacker/go-tree-sitter/golang"
+	"github.com/smacker/go-tree-sitter/hcl"
+	"github.com/smacker/go-tree-sitter/html"
+	"github.com/smacker/go-tree-sitter/java"
+	"github.com/smacker/go-tree-sitter/javascript"
+	"github.com/smacker/go-tree-sitter/lua"
+	"github.com/smacker/go-tree-sitter/ocaml"
+	"github.com/smacker/go-tree-sitter/php"
+	"github.com/smacker/go-tree-sitter/protobuf"
+	"github.com/smacker/go-tree-sitter/python"
+	"github.com/smacker/go-tree-sitter/ruby"
+	"github.com/smacker/go-tree-sitter/rust"
+	"github.com/smacker/go-tree-sitter/scala"
+	"github.com/smacker/go-tree-sitter/svelte"
+	"github.com/smacker/go-tree-sitter/toml"
+	"github.com/smacker/go-tree-sitter/typescript/typescript"
+	"github.com/smacker/go-tree-sitter/yaml"
+)
+
+// ParsedIssue represents an issue parsed using tree-sitter.
+type ParsedIssue struct {
+	IssueCode string
+	Line      int
+}
+
+type ParsedIssues []ParsedIssue
+
+func Run(directory string) error {
+	// read the generated report from TOOLBOX_PATH
+	toolboxPath := os.Getenv("TOOLBOX_PATH")
+	generatedFile := path.Join(toolboxPath, "analysis_report.json")
```

Comment:
What happens if this file does not exist?

Reply:
It should exit with an error.

`analysistest.Run()` is meant for testing analyzers; if the file (report) doesn't exist, it exits the runner with an error. The error can be handled inside the tests using `err = analysistest.Run()`.

Reply:
Ack.